mono/metadata/reflection.c

```diff
@@ -3090,3 +3090,20 @@ mono_class_from_mono_type_handle (MonoReflectionTypeHandle reftype)
 {
 	return mono_class_from_mono_type_internal (MONO_HANDLE_RAW (reftype)->type);
 }
+
+// This is an icall, it will return NULL and set pending exception on failure.
+MonoReflectionType*
+mono_type_from_handle (MonoType *handle)
```

Comment:
So this is like `mono_type_get_object_{checked,handle}`, except that it also calls `mono_class_init_internal` - it would be better if it followed their naming scheme. (Also I'm not a fan of hiding pending exception setting in helper functions)

Reply:
I agree the functions need work but first I wanted to combine them w/o change.